storage: TestPebbleIterator_ExternalCorruption failed

storage.TestPebbleIterator_ExternalCorruption [failed](https://tanzanite.cluster.engflow.com/invocations/default/bdd258af-3d5a-47a8-b192-7e432ea4adae?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3RvcmFnZTpzdG9yYWdlX3Rlc3Q=) on release-25.1.0-rc @ [76c2bc942ee1d8d8d62e32047bcf3cacaa21fdc1](https://github.com/cockroachdb/cockroach/commits/76c2bc942ee1d8d8d62e32047bcf3cacaa21fdc1):

Fatal error:

```
panic: pebble/table: 000000: unknown key schema ""; known key schemas: ["crdb1"] [recovered]
	panic: pebble/table: 000000: unknown key schema ""; known key schemas: ["crdb1"] [recovered]
	panic: pebble/table: 000000: unknown key schema ""; known key schemas: ["crdb1"]

```
Stack: 

```
goroutine 66181 [running]:
testing.tRunner.func1.2({0x28ed740, 0xc00014b3f8})
	GOROOT/src/testing/testing.go:1631 +0x24a
testing.tRunner.func1()
	GOROOT/src/testing/testing.go:1634 +0x377
panic({0x28ed740?, 0xc00014b3f8?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func2()
	pkg/util/leaktest/leaktest.go:132 +0x3aa
panic({0x28ed740?, 0xc00014b3f8?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/pebble/sstable.NewReader({0x3123888, 0x4ba7d40}, {0x3124310, 0xc002458e40?}, {{{0x0, 0x0, 0x0}, 0x0, {0x3129080, 0x4ba7d40}}, ...})
	external/com_github_cockroachdb_pebble/sstable/reader.go:872 +0xf7d
github.com/cockroachdb/pebble.openExternalTables({0x3123888, 0x4ba7d40}, 0x4459e5?, {0xc001d1deb0, 0x1, 0xc001d1d4e8?}, 0x445b1c?, {{{0x0, 0x0, 0x0}, ...}, ...})
	external/com_github_cockroachdb_pebble/external_iterator.go:272 +0x10e
github.com/cockroachdb/pebble.NewExternalIterWithContext({0x3123888, 0x4ba7d40}, 0xc00303d508, 0xc001bdfdc8, {0xc001d1dec0?, 0x1?, 0x2?})
	external/com_github_cockroachdb_pebble/external_iterator.go:59 +0x225
github.com/cockroachdb/pebble.NewExternalIter(...)
	external/com_github_cockroachdb_pebble/external_iterator.go:37
github.com/cockroachdb/cockroach/pkg/storage.newPebbleSSTIterator({0xc001d1dec0, 0x1, 0x1}, {0x0, {0x0, 0x0, 0x0}, {0x485411c, 0x2, 0x2}, ...})
	pkg/storage/pebble_iterator.go:139 +0xf2
github.com/cockroachdb/cockroach/pkg/storage.NewSSTIterator(...)
	pkg/storage/sst.go:44
github.com/cockroachdb/cockroach/pkg/storage.TestPebbleIterator_ExternalCorruption(0xc00222e340)
	pkg/storage/pebble_iterator_test.go:127 +0x3f7
testing.tRunner(0xc00222e340, 0x2cb9cb8)
	GOROOT/src/testing/testing.go:1689 +0xfb
created by testing.(*T).Run in goroutine 1
	GOROOT/src/testing/testing.go:1742 +0x390
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestPebbleIterator_ExternalCorruption
    reader.go:872: panic: pebble/table: 000000: unknown key schema ""; known key schemas: ["crdb1"]
--- FAIL: TestPebbleIterator_ExternalCorruption (0.00s)
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/storage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPebbleIterator_ExternalCorruption.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47846

--- a/pkg/storage/pebble_iterator_test.go
+++ b/pkg/storage/pebble_iterator_test.go
@@ -122,7 +122,14 @@ func TestPebbleIterator_ExternalCorruption(t *testing.T) {
 
 	// Trash a random byte.
 	b := f.Bytes()
-	b[rng.Intn(len(b))]++
+
+	// If we mess with the format byte, we will get an unexpected error.
+	// See https://github.com/cockroachdb/cockroach/issues/141477 and
+	// TODO(radu): This can be removed if Pebble checksums the footer:
+	// https://github.com/cockroachdb/pebble/issues/4344
+	const nAvoidLastBytes = 12
+	//b[rng.Intn(len(b)-nAvoidLastBytes)]++
+	b[len(b)-60+rng.Intn(60-nAvoidLastBytes)]++
 
 	it, err := NewSSTIterator([][]sstable.ReadableFile{{vfs.NewMemFile(b)}},
 		IterOptions{UpperBound: roachpb.KeyMax})

